### PR TITLE
Allow to turn off searching subtitles for specials globally

### PR DESCRIPTION
**Pull request by sharkykh**
I wrote this code about a week ago, and it was just laying around on a branch.
Figured I might as well open a PR for it.

I know SickRage doesn't really handle specials, but every once in a while it decides that it downloads a subtitle for a show's special episode.

I found places in the code where it specifically checks `season != 0` to exclude specials,
and others where no check exists.
It's better to have it as a checkbox, no?

* Adds an `Include Specials` option to Subtitles Config.
* Standardises subtitle searching for specials throughout SickRage:
  * Find Subtitles thread
  * Specials in `Missed Subtitles`
  * cc next to special episodes on show page

You/We should decide what is the default state of that option...
I put `True` because it's closer to what we have right now.

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

File: gui/slick/views/displayShow.mako [hunk 1]
```diff
@@ -556,8 +556,10 @@
...
                                                 % endif
                                             % endif
-                                            % if int(epResult[b"status"]) not in Quality.SNATCHED + Quality.SNATCHED_PROPER and sickbeard.USE_SUBTITLES and show.subtitles and epResult[b"location"] and subtitles.needs_subtitles(epResult[b'subtitles']):
-                                                <a class="epSubtitlesSearch" href="searchEpisodeSubtitles?show=${show.indexerid}&amp;season=${epResult[b"season"]}&amp;episode=${epResult[b"episode"]}"><span class="displayshow-icon-sub" title="Search Subtitles" /></a>
+                                            % if int(epResult[b"status"]) not in Quality.SNATCHED + Quality.SNATCHED_PROPER and sickbeard.USE_SUBTITLES and show.subtitles and epResult[b"location"] and subtitles.needs_subtitles(epResult['subtitles']):
+                                                % if int(epResult[b"season"]) != 0 or (int(epResult[b"season"]) == 0 and sickbeard.SUBTITLES_INCLUDE_SPECIALS):
```
Comment (miigotu) [inline gui/slick/views/displayShow.mako]: `if int(epResult[b"season"]) != 0 or sickbeard.SUBTITLES_INCLUDE_SPECIALS:`

File: sickbeard/postProcessor.py [hunk 2]
```diff
@@ -1179,7 +1179,8 @@ def process(self):  # pylint: disable=too-many-return-statements, too-many-local
             with cur_ep.lock:
                 cur_ep.location = ek(os.path.join, dest_path, new_file_name)
                 # download subtitles
-                if sickbeard.USE_SUBTITLES and ep_obj.show.subtitles:
+                if sickbeard.USE_SUBTITLES and ep_obj.show.subtitles \
+                    and (cur_ep.season != 0 or cur_ep.season == 0 and sickbeard.SUBTITLES_INCLUDE_SPECIALS):
```
Comment (miigotu) [inline sickbeard/postProcessor.py]: `and (cur_ep.season != 0 or sickbeard.SUBTITLES_INCLUDE_SPECIALS):`

**Review by miigotu — CHANGES_REQUESTED** [review]

**Comment by sharkykh** [comment]
### You might want to add this fix too
```
AA IndexError: Index must be int or string
AA subtitle_path = ep_to_sub['location'].encode(sickbeard.SYS_ENCODING)
AA File "C:\SickRage\SickRage\sickbeard\subtitles.py", line 445, in run
AA self.action.run(self.force)
AA File "C:\SickRage\SickRage\sickbeard\scheduler.py", line 109, in run
AA Traceback (most recent call last):
2017-02-28 04:35:47 ERROR FINDSUBTITLES :: [14e02c4] Exception generated in thread FINDSUBTITLES: Index must be int or string
```

**Comment by miigotu** [comment]
That sqlit3.Row quirk is annoying, I wish I could make it autoencode to utf-8 for column lookups lol.

**Review by miigotu — APPROVED** [review]

**Pull request merged**